api/src/routers/meals.js

```diff
@@ -74,4 +74,64 @@ mealsRouter.delete("/:id", async (req, res, next) => {
   }
 });
 
+mealsRouter.get("/", async (request, response, next) => {
+  console.log("GET");
+  try {
+    const query = db("Meals");
+    const {
+      maxPrice,
+      availableReservations,
+      title,
+      dateAfter,
+      dateBefore,
+      limit,
+      sortKey,
+      sortDir,
+    } = request.query;
+
+    if (maxPrice !== undefined) {
+      query.where("price", "<", maxPrice);
+    }
+    if (req.query.availableReservations) {
+      const available = req.query.availableReservations === "true";
+      query = query
+        .leftJoin("reservation", "meal.id", "reservation.meal_id")
+        .groupBy("meal.id")
+        .select("meal.id", "meal.title", "meal.price", "meal.max_reservations")
+        .having(
+          `meal.max_reservations - COUNT(reservation.id) ${available ? ">" : "="} 0`
+        );
+    }
+    if (title !== undefined) {
+      query.where("title", "like", `%${title}%`);
+    }
+
+    if (dateAfter !== undefined) {
+      query.where("when", ">", dateAfter);
+    }
+    if (dateBefore !== undefined) {
+      query.where("when", "<", dateBefore);
+    }
+    if (limit !== undefined) {
+      query.limit(limit);
+    }
+    if (sortKey !== undefined) {
+      if (sortKey == "price") {
+        query.orderBy("price", sortDir !== undefined ? sortDir : "asc");
+      }
+      if (sortKey == "max_reservations") {
+        query.orderBy(
+          "max_reservations",
+          sortDir !== undefined ? sortDir : "asc"
+        );
+      }
+    }
```

Comment:
Hi you are missing validation of these fields. As a user of your endpoint it can be frustrating to not be told if your request was wrong :) For example, limit has to be an integer, if the request is not an integer they should be told so with an error code 400